taskcluster/kinds/clean-corpus/kind.yml

```diff
@@ -68,7 +68,11 @@ tasks:
                 USE_OPUSCLEANER: "{use_opuscleaner}"
                 COMPRESSION_CMD: zstdmt
                 ARTIFACT_EXT: zst
-    
+
+# TODO: doesn't work. How to enable retries when the task fails?
+#        # there might be intermittent issues with OpusCleaner, specifically FastText model downloading can fail
+#        retries: 5
```

Comment:
@bhearsum this didn't work for me. I see in the docs that task retries work only when infrastructure fails. How to add retries when the task fails?

Reply:
I think you can use `retry-exit-status` for this: https://github.com/taskcluster/taskgraph/blob/a7345114e4af294dc31c22c438a5ddb3467676da/src/taskgraph/transforms/task.py#L349.

Another option would be to retry within the command executed by the task. It might be better: the retry would work independently whether the command is run on Taskcluster or locally, and it would be possible to have a smarter retry algorithm (e.g. exponential backoff).

Reply:
Also, by retrying the download within the command, you don't "waste" the time that has elapsed so far (e.g. installing packages, etc.).

Reply:
There are multiple ways to add retries but it should be supported by the workflow manager. It is a standard functionality. See [snakemake](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#defining-retries-for-fallible-rules) for example. Rather than adding code in each script that has external dependencies or other intermittent issues I'd set one line in Taskcluster definitions. And we have external dependencies in many tasks (datasets, cleaning, bicleaner, pip install, github etc.).

Reply:
If we want to be economical about spinning up the machines, we should move to docker images where everything is already installed and cached. Some installations can take quite a while.

Reply:
It is supported and you can use it. I was suggesting not to use it because I think it might better to do it explicitly within the task when it makes sense than at the task level.
Note even if you use Docker where everything is installed and cached, you're still going to pay the price of spinning up a machine and downloading the Docker image from a registry.

That said, it is just an opinion about a small thing, and it's not a strong opinion, so feel free to do what you want ;)

Reply:
I see, I thought you're saying it's not supported. I'd rather reduce complexity of the code by handling it on the task level. This should not happen often. If it does we'll want to address this particular problem differently.